web/assignments.go

```diff
@@ -74,6 +79,55 @@ func (s *AutograderService) deleteCriterion(query *pb.GradingCriterion) error {
 	return s.db.DeleteCriterion(query)
 }
 
+func (s *AutograderService) loadCriteria(ctx context.Context, sc scm.SCM, request *pb.LoadCriteriaRequest) ([]*pb.GradingBenchmark, error) {
+
+	assignment, err := s.db.GetAssignment(&pb.Assignment{ID: request.AssignmentID, CourseID: request.CourseID})
+	if err != nil {
+		return nil, err
+	}
+
+	course, err := s.db.GetCourse(request.CourseID, false)
+	if err != nil {
+		return nil, err
+	}
+
+	opts := &scm.FileOptions{
+		Path:       filepath.Join(assignment.GetName(), criteriaFile),
+		Owner:      course.OrganizationPath,
+		Repository: pb.TestsRepo,
+	}
+
+	criteriaString, err := sc.GetFileContent(ctx, opts)
+	if err != nil || criteriaString == "" {
```

Comment:
What if `err == nil` and `criteriaString == ""`. This method will return `nil, nil`. Is that expected?

Reply:
Some would argue that here one should return an empty `[]*pb.GradingBenchmark`. Or maybe it should be an error if the file content is empty?

Reply:
Right. I think there should be an error in this case, it will now be returned by the `GetFileContent` method if the file is empty and logged properly.

Reply:
Looks good to me!